packages/connect-query/src/unary-hooks.ts

```diff
@@ -106,14 +106,40 @@ export interface UnaryHooks<I extends Message<I>, O extends Message<O>> {
    */
   useInfiniteQuery: <ParamKey extends keyof PartialMessage<I>>(
     input: DisableQuery | PartialMessage<I>,
-    options: {
-      pageParamKey: ParamKey;
-      getNextPageParam: (lastPage: O, allPages: O[]) => unknown;
-
-      onError?: (error: ConnectError) => void;
-      transport?: Transport | undefined;
-      callOptions?: CallOptions | undefined;
-    },
+    options:
+      | {
+          applyPageParam: (options: {
+            pageParam: PartialMessage<I>[ParamKey] | undefined;
+            input: PartialMessage<I>;
+          }) => PartialMessage<I>;
+          getNextPageParam: (
+            lastPage: O,
+            allPages: O[],
+          ) => PartialMessage<I>[ParamKey];
+          /**
+           * The option allows you to remove fields or otherwise customize the input used to generate the query key.
+           * By default, we will remove the pageParamKey from the input. If this is provided, we will use this result instead.
+           */
+          sanitizeInputKey?: (input: PartialMessage<I>) => unknown;
+          onError?: (error: ConnectError) => void;
+          transport?: Transport | undefined;
+          callOptions?: CallOptions | undefined;
+        }
+      | {
+          pageParamKey: ParamKey;
+          getNextPageParam: (
+            lastPage: O,
+            allPages: O[],
+          ) => PartialMessage<I>[ParamKey];
+          /**
+           * The option allows you to remove fields or otherwise customize the input used to generate the query key.
+           * By default, we will remove the pageParamKey from the input. If this is provided, we will use this result instead.
+           */
+          sanitizeInputKey?: (input: PartialMessage<I>) => unknown;
+          onError?: (error: ConnectError) => void;
+          transport?: Transport | undefined;
+          callOptions?: CallOptions | undefined;
+        },
```

Comment:
There is probably a better way to do this, but moving it to a shared interface makes it so type assertions on the param result seems to break without exposing those shared types.